### PR TITLE
MXLIFF: Fixed CU6 error.

### PR DESCRIPTION
**Pull request by andreaGhisa**
Fixed CU6 path not find issue.
Added null validations
Added maxversion attribute.

### CODE REVIEW

File: FileTypeSupport.MXLIFF/MXLIFFWriter.cs [hunk 1]
```diff
@@ -33,109 +32,107 @@ public void Dispose()
...
+	        _documentProperties = documentInfo;
+            _textExtractor = new MXLIFFTextExtractor();
         }
 
         public void ProcessParagraphUnit(IParagraphUnit paragraphUnit)
         {
-            string id = paragraphUnit.Properties.Contexts.Contexts[0].GetMetaData("ID");
-            XmlNode xmlUnit = targetFile.SelectSingleNode("//x:trans-unit[@id='" + id + "']", nsmgr);
+            var id = paragraphUnit.Properties.Contexts.Contexts[0].GetMetaData("ID");
+            var xmlUnit = _targetFile.SelectSingleNode("//x:trans-unit[@id='" + id + "']", _nsmgr);
```
Comment (TinaCaputa) [inline FileTypeSupport.MXLIFF/MXLIFFWriter.cs:56]: check for null validation on _targetFile

Comment (andreaGhisa) [reply FileTypeSupport.MXLIFF/MXLIFFWriter.cs:56]: It doessn't need to is always instantiated in line 63.

File: FileTypeSupport.MXLIFF/MXLIFFWriter.cs [hunk 2]
```diff
@@ -33,109 +32,107 @@ public void Dispose()
...
-                    var id = user.Attributes["id"]?.Value;
-                    var username = user.Attributes["username"]?.Value;
-
-                    if (!string.IsNullOrWhiteSpace(username) && !string.IsNullOrWhiteSpace(id) && users.ContainsKey(username))
-                    {
-                        users.Add(username, id);
-                    }
-                }
-            }
+            var memsourceUsers = _targetFile.SelectNodes("//m:user", _nsmgr);
```
Comment (TinaCaputa) [inline FileTypeSupport.MXLIFF/MXLIFFWriter.cs:81]: Is it possible that _targetFile will be null?

File: FileTypeSupport.MXLIFF/MXLIFFWriter.cs [hunk 3]
```diff
@@ -149,50 +146,41 @@ private void AddComments(XmlNode xmlUnit, List<IComment> comments)
                 text += c.Text + " ";
             }
 
-            var createdat = targetFile.CreateAttribute("created-at");
-            var createdby = targetFile.CreateAttribute("created-by");
-            var modifiedat = targetFile.CreateAttribute("modified-at");
+            var createdat = _targetFile.CreateAttribute("created-at");
```
Comment (TinaCaputa) [inline FileTypeSupport.MXLIFF/MXLIFFWriter.cs:149]: Is it possible that _targetFile value is null?

File: FileTypeSupport.MXLIFF/MXLIFFWriter.cs [hunk 4]
```diff
@@ -149,50 +146,41 @@ private void AddComments(XmlNode xmlUnit, List<IComment> comments)
...
             // If it fails, just leave it blank
-            if (users.ContainsKey(comment.Author))
-            {
-                createdby.Value = users[comment.Author].ToString();
-            }
-            else
-            {
-                createdby.Value = "";
-            }
+            createdby.Value = _users.ContainsKey(comment.Author) ? _users[comment.Author] : "";
```
Comment (TinaCaputa) [inline FileTypeSupport.MXLIFF/MXLIFFWriter.cs:165]: Use string.Empty instead of ""

File: FileTypeSupport.MXLIFF/MXLIFFWriter.cs (hunk 2 above)
Comment (andreaGhisa) [reply FileTypeSupport.MXLIFF/MXLIFFWriter.cs:81]: Is instanciated in line 63.

File: FileTypeSupport.MXLIFF/MXLIFFWriter.cs (hunk 3 above)
Comment (andreaGhisa) [reply FileTypeSupport.MXLIFF/MXLIFFWriter.cs:149]: Is instanciated in line 63.

**Pull request merged**